src/Components/Assets/AssetImportModal.tsx

```diff
@@ -31,24 +32,28 @@ const AssetImportModal = ({ open, onClose, facility }: Props) => {
     useState<(AssetData & { notes?: string; last_serviced_on?: string })[]>();
   const [location, setLocation] = useState("");
   const [locations, setLocations] = useState<any>([]);
-  const dispatchAction: any = useDispatch();
   const { sample_format_asset_import } = useConfig();
 
   const closeModal = () => {
     setPreview(undefined);
     setSelectedFile(undefined);
     onClose && onClose();
   };
+  const {
+    data: facilityAssetLocations,
+    loading,
+    refetch,
+  } = useQuery(routes.listFacilityAssetLocation, {
+    pathParams: { facility_external_id: `${facility.id}` },
+  });
 
   useEffect(() => {
-    dispatchAction(
-      listFacilityAssetLocation({}, { facility_external_id: facility.id })
-    ).then(({ data }: any) => {
-      if (data.count > 0) {
-        setLocations(data.results);
-      }
-    });
-  }, []);
+    if (!facilityAssetLocations) {
+      () => refetch();
+    } else if (facilityAssetLocations?.count) {
+      setLocations(facilityAssetLocations?.results);
+    }
```

Comment:
You can chain multiple useQueries by passing `prefetch=false` to all dependent queries and call it's refetch once data is available.